delta_filter.cpp

```diff
@@ -217,56 +273,92 @@ struct timeval	now, res;
 								(*nIt)->getName().c_str());
 				}
 			}
-			else
+			else // DPV type hasn't changed
 			{
-
 				switch(nValue.getType())
 				{
-					case DatapointValue::T_INTEGER:
-						if (abs(nValue.toInt() - oValue.toInt())
-							> (tolerance * abs(oValue.toInt())) / 100)
-						{
-							sendThis = true;
-						}
-						break;
-					case DatapointValue::T_FLOAT:
-						if (fabs(nValue.toDouble() - oValue.toDouble())
-							> (tolerance * fabs(oValue.toDouble())) / 100)
-						{
-							sendThis = true;
-						}
-						break;
-					case DatapointValue::T_STRING:
-						if (nValue.toString().compare(oValue.toString()))
-						{
-							sendThis = true;
-						}
-						break;
-					case DatapointValue::T_FLOAT_ARRAY:
-						// T_FLOAT_ARRAY not supported right now
-					default:
-						break;
-				}
-			}
-			if (sendThis)
-			{
-				break;
-			}
-		}
-		if (sendThis)
-		{
-			break;
-		}
-	}
+                    case DatapointValue::T_INTEGER:
+                    case DatapointValue::T_FLOAT:
+                        {
+                            bool toleranceExceeded = checkToleranceExceeded(oValue, nValue, toleranceMeasure, tolerance, absChange);
+                            if (toleranceExceeded)
+                            {
+                                Logger::getLogger()->info("Datapoint %s has %lf %schange", (*nIt)->getName().c_str(), absChange,
+                                                            (toleranceMeasure == ToleranceMeasure::PERCENTAGE)? "% " : "");
+                                changedDPs.emplace((*nIt)->getName());
+                            }
+                        }
+                        break;
+
+                    case DatapointValue::T_STRING:
+                        {
+                            bool toleranceExceeded = checkToleranceExceeded(oValue, nValue, toleranceMeasure, tolerance, absChange);
+                            if (toleranceExceeded)
+                            {
+                                Logger::getLogger()->info("Datapoint %s of STRING type has changed from '%s' to '%s'", 
+                                                            (*nIt)->getName().c_str(), oValue.toString().c_str(), nValue.toString().c_str());
+                                changedDPs.emplace((*nIt)->getName());
+                            }
+                        }
+                        break;
+
+                    case DatapointValue::T_FLOAT_ARRAY:
+                        // T_FLOAT_ARRAY not supported right now
+                    
+                    default:
+                        break;
+                }
+            }
+        }
+    }
+
+    Logger::getLogger()->info("processingMode=%d, changedDPs.size()=%d, nDataPoints.size()=%d", 
+                                processingMode, changedDPs.size(), nDataPoints.size());
 
-	if (sendThis)
+    // Act according to processingMode config
+	if ( maxPeriodElapsed ||
```

Comment:
It's not clear what decision is being taken here on the basis of processing mode.

Reply:
Whether to send the whole reading or reading with only a subset of DPs.

Reply:
Added detailed comment.